tests/unit/igraph_steiner_tree_fpt.c

```diff
@@ -0,0 +1,118 @@
+
+int main()
+{
+   // printf("Starting Tests\n");
+    igraph_t g_empty, g_lm;
+
+    igraph_vector_int_t steiner_terminals,steiner_terminals_null;
+    igraph_vector_t weights_empty, weights_lm;
+    
+    igraph_vector_init(&weights_empty, 0);
+
+    igraph_vector_int_init(&steiner_terminals, 4);
+    igraph_vector_int_init(&steiner_terminals_null, 0);
+
+    VECTOR(steiner_terminals)[0] = 0;
+    VECTOR(steiner_terminals)[1] = 1;
+    VECTOR(steiner_terminals)[2] = 2;
+    VECTOR(steiner_terminals)[3] = 3;
+
+    igraph_vector_init(&weights_lm, 21);
+    
+    VECTOR(weights_lm)[0] = 2;
+    VECTOR(weights_lm)[1] = 2;
+    VECTOR(weights_lm)[2] = 2;
+    VECTOR(weights_lm)[3] = 1;
+    VECTOR(weights_lm)[4] = 1;
+    VECTOR(weights_lm)[5] = 2;
+    
+    VECTOR(weights_lm)[6] = 2;
+    VECTOR(weights_lm)[7] = 2;
+    VECTOR(weights_lm)[8] = 2;
+    VECTOR(weights_lm)[9] = 1;
+    VECTOR(weights_lm)[10] = 2;
+
+    VECTOR(weights_lm)[11] = 2;
+    VECTOR(weights_lm)[12] = 2;
+    VECTOR(weights_lm)[13] = 2;
+    VECTOR(weights_lm)[14] = 1;
+    
+    VECTOR(weights_lm)[15] = 1;
+    VECTOR(weights_lm)[16] = 2;
+    VECTOR(weights_lm)[17] = 1;
+
+    VECTOR(weights_lm)[18] = 2;
+    VECTOR(weights_lm)[19] = 1;
+
+    VECTOR(weights_lm)[20] = 1;
+
+    igraph_empty(&g_empty, 0, 0);
+
+    igraph_small(&g_lm, 7, IGRAPH_UNDIRECTED, 
+                                                0, 1,
+                                                0, 2,
+                                                0, 3, 
+                                                0, 4, 
+                                                0, 5, 
+                                                0, 6, 
+                                                
+                                                1, 2, 
+                                                1, 3, 
+                                                1, 4, 
+                                                1, 5, 
+                                                1, 6,
+
+                                                2, 3, 
+                                                2, 4, 
+                                                2, 5, 
+                                                2, 6, 
+                                                
+                                                3, 4, 
+                                                3, 5, 
+                                                3, 6, 
+                                                
+                                                4, 5, 
+                                                4, 6, 
+                                                
+                                                5, 6, 
+                                                -1);
+    
+    //printf("No vertices, not directed:\n");
+    IGRAPH_ASSERT(igraph_steiner_dreyfus_wagner(&g_empty,&steiner_terminals_null, IGRAPH_UNDIRECTED, &weights_empty) == IGRAPH_SUCCESS);
```

Comment:
Tip: you don't need to assert a return of `IGRAPH_SUCCESS` in tests _unless_ you changed the default error handler. If there is an error, the program will abort by default, and the test will fail.

Reply:
Okay. I had looked into other test cases and since they were using it, I thought it is necessary. But I will make sure to change it